app/src/app/app.component.spec.ts

```diff
@@ -1,32 +1,40 @@
-import { TestBed, async } from '@angular/core/testing';
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
 
+import { EnplugService } from './enplug.service';
 import { AppComponent } from './app.component';
 
+let enplugServiceMock: any;
+let fixture: ComponentFixture<AppComponent>;
+
 describe('AppComponent', () => {
-  beforeEach(async(() => {
+  beforeEach(() => {
+    enplugServiceMock = {
+      appStatus: jasmine.createSpyObj('appStatus', ['start']),
+      on: jasmine.createSpy('on')
+    };
+
     TestBed.configureTestingModule({
-      declarations: [
-        AppComponent
-      ],
+      providers: [{ provide: EnplugService, useValue: enplugServiceMock }],
+      declarations: [AppComponent],
     }).compileComponents();
-  }));
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AppComponent);
+  });
 
   it('should create the app', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
     const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
+    expect(app).toBeDefined();
   }));
 
-  it(`should have as title 'app'`, async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app.title).toEqual('app');
+  it('calls Player\'s start function on init', async(() => {
```

Comment:
`enplug.appStatus.start()` doesn't have to be called in that particular moment, just in a few seconds after the app loads. Let's make this test wait (let's say) 5s and then check for `start` to have been called.

Reply:
5 seconds seems a lot and it's default time jasmine waits for single test to finish.
I'd rather set it to lower number, not to delay test suite much. Maybe 2 seconds?

Reply:
I think calling appStatus.start(); depends on an app -> in some cases it might be instant in other cases it might take a few seconds. I'm in favor of leaving it as an instant check, so the AppSeed would pass and if the developer modifies the app in a way that would call the appStatus.start() a bit later, the test would fail and that would make the developer to adapt the test to the app that he's developing.